libbeat/template/field.go

```diff
@@ -23,6 +23,7 @@ type Field struct {
 	Analyzer       string `config:"analyzer"`
 	SearchAnalyzer string `config:"search_analyzer"`
 	Norms          bool   `config:"norms"`
+	Dynamic        string `config:"dynamic"`
```

Comment:
I'm missing some error handling for invalid strings.

e.g.

```

type Field struct {
  ...
  Dynamic dynamicType
  ...
}

type dynamicType interface{}

func (d *dynamicType) Unpack(s string) error {
  switch s {
    case "true":
      *d = true
    case "false":
      *d = false
    case "strict":
      *d = s
    default:
      return fmt.Errorf("'%v' is invalid dynamic setting", s)
  }
  return nil
}
```

you can also support `bool` in general via `interface{}`:

```
func (d *dynamicType) Unpack(in interface{}) error {
  if b, ok := in.(bool); ok {
    *d = b
    return nil
  }

  s, ok := in.(string)
  if !ok {
    return fmt.Errorf(...)
  }

  switch s {
    case "true":
      *d = true
    case "false":
      *d = false
    case "strict":
      *d = s
    default:
      return fmt.Errorf("'%v' is invalid dynamic setting", s)
  }
  return nil
}
```